docs/source/http_api.rst

```diff
@@ -1,5 +1,7 @@
 HTTP REST API
 =============
 
-.. autoflask:: topchef.api_server:app
+This document contains the documentation for the HTTP API.
+
+.. autoflask:: topchef.wsgi_app:APP_FACTORY.app
```

Comment:
I'm so glad that this worked so seamlessly with the metaclasses we designed earlier. 

When the ``sphinxcontrib-httpdomain`` hits that app just right

![pacha](https://user-images.githubusercontent.com/8391612/29368069-b1a2ac9a-826c-11e7-917a-35923433c29c.jpg)